share/ansible.sh

```diff
@@ -0,0 +1,25 @@
+#!/bin/bash
+#
+# Copyright 2021, Marshall McMullen <marshall.mcmullen@gmail.com>
+#
+# This program is free software: you can redistribute it and/or modify it under the terms of the Apache License
+# as published by the Apache Software Foundation, either version 2 of the License, or (at your option) any later
+# version.
+
+opt_usage ansible_status <<'END'
+`ansible_status` is used to relay status back to Ansible in expected JSON format for completed tasks.
+This prints status for Ansible in the following format -
+{"failed": <true|false> "changed": <true|false> "msg": <user provided message>}
+END
+ansible_status()
+{
+    $(opt_parse \
+        "=failed    f   | Did this module fail - true | false."             \
+        "=changed   c   | Did this module change anything - true | false."  \
+        "=msg       m   | Message associated with this status.")
+
+    # Note - Ansible is particular about it's status output. We need to escape
+    #        any characters with special meaning in JSON.
+    msg=$(json_escape "${msg}")
```

Comment:
WOO! Thank you for using `json_escape` here instead of pulling in python dependencies.